src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -49,6 +50,15 @@
     private static final String ISO_8601 = "yyyy-MM-dd'T'HH:mm:ss";
     private static final String ISO_8601_WITH_TZ = "yyyy-MM-dd'T'HH:mm:ssX";
 
+    // Nearly ISO dates generated by git whatchanged --format=+ci
+    // Look like '2015-09-30 08:21:24 -0600'
+    // ISO is    '2015-09-30T08:21:24-06:00'
+    private static final Pattern NEARLY_ISO_DATE = Pattern.compile("^([0-9]{4}-[0-9]{2}-[0-9]{2})" + // YYYY-MM-DD
```

Comment:
Thanks.  The test writing effort ultimately exposed a very real surprise in
at least two older command line git implementations (CentOS 6 and Debian 6
git versions have an issue with some calls to "git whatchanged").  Those
two older implementations are not fully supported by the plugin, since they
don't include credentials support, but they are known to work and I like to
understand things that might break before users report them as broken.

I've closed this pull request opened another, and merged that pull request,
along with extra tests of the cases that I detected.  Many hours of work
trying to assure that a timestamp will be reported correctly through the
API.

Mark Waite

On Sat, Oct 17, 2015 at 7:24 AM, Oleg Nenashev <notifications@github.com>
wrote:

> In src/main/java/hudson/plugins/git/GitChangeSet.java
> <https://github.com/jenkinsci/git-plugin/pull/355#discussion_r42309234>:
>
> > @@ -49,6 +50,15 @@
> >      private static final String ISO_8601 = "yyyy-MM-dd'T'HH:mm:ss";
> >      private static final String ISO_8601_WITH_TZ = "yyyy-MM-dd'T'HH:mm:ssX";
> >
> > +    // Nearly ISO dates generated by git whatchanged --format=+ci
> > +    // Look like '2015-09-30 08:21:24 -0600'
> > +    // ISO is    '2015-09-30T08:21:24-06:00'
> > +    private static final Pattern NEARLY_ISO_DATE = Pattern.compile("^([0-9]{4}-[0-9]{2}-[0-9]{2})" + // YYYY-MM-DD
>
> @MarkEWaite <https://github.com/MarkEWaite>
> Sorry for the missing response. Yes, my concern was about decreasing the
> complexity and technical debt by relying on standard implementations if
> they are feasible. Thanks for taking the action!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/git-plugin/pull/355/files#r42309234>.
>



-- 
Thanks!
Mark Waite